### ISSUE TITLE
concurrency not working with helmfiles pattern

### ISSUE THREAD

**Issue by FACOLOMBANI** [helmfile/helmfile#528]
### Operating system

Ubuntu 20.04, Mac OS Monterrey 12.6.1

### Helmfile Version

0.148.1

### Helm Version

3.10.2

### Bug description

I have a definition where i have a helmfile per component and a global helmfile specifying the path to the different component helmfiles as follow:

```
helmfiles:
- path: "releases/first/helmfile.yaml"
- path: "releases/second/helmfile.yaml"
- path: "releases/third/helmfile.yaml"
```

The problem i am encountering is that each helmfile is being deploy in sequence, 1 by 1, waiting each time for the component to be ready before deploying next helmfile.

I would expect to have the same behaviour as with releases definition where all components are deployed concurrently:
```
releases:
- name: first
- name: second
- name: third
```

NOTE: concurrent flag does not work.

### Example helmfile.yaml

```
helmfiles:
- path: "releases/first/helmfile.yaml"
- path: "releases/second/helmfile.yaml"
- path: "releases/third/helmfile.yaml"
```

### Error message you've seen (if any)

No error message just the behaviour of a sequential deployment instead of concurrent deployments.

### Steps to reproduce

https://github.com/FACOLOMBANI/helmfile-test

### Working Helmfile Version

Problem is with latest version. Version 0.148.1

**Comment by yxxhero** [comment]
ping pong

**Comment by FACOLOMBANI** [comment]
Any news about this topic? it was tagged as enhancement and now to won't fix...

**Comment by yxxhero** [comment]
@FACOLOMBANI I will try to see this issue.

**Comment by yxxhero** [comment]
@FACOLOMBANI it's expected to behave.
1. if you want concurrent deployments. please input all releases into a helmfile.
2. you can control the release order by order of `helmfile`  in `helmfiles`.


@mumoshu @itscaro WDYT? I think it's normal. or maybe we can add a flag to control it.

**Comment by itscaro** [comment]
It's by design, and stated in the documentation with the use-case of CRD as well https://github.com/helmfile/helmfile/blob/398c812e4902f389eb97717216b837604d050eb3/docs/index.md?plain=1#L352-L353

Implementing a choice between sequential or parrallel might add more complexity to helmfile?

If you need to run releases in parallel and still want to have small helmfiles, maybe you can add a pre-process to build with `helmfile build` and merge all releases into a new document and pipe to `helmfile apply`.

**Comment by yxxhero** [comment]
@itscaro it is not easy to implement it.

**Comment by FACOLOMBANI** [comment]
I did the test but the result is the same, perhaps i am missing something. This is what i did:
1. helmfile build into a new document 
2. helmfile apply of this new document

**Comment by jouve** [comment]
you might try something like this to produce a merged helmfile:
```
helmfile build | yq ea 'select(.releases) | . as $item ireduce({}; . *+ $item) | .repositories = (.repositories | unique_by("name"))' 
```

**Comment by jouve** [comment]
for example in other projects, argo workflow use nested lists to specify steps to run in parallel: https://argoproj.github.io/argo-workflows/walk-through/steps/

maybe hemfile could evolve to support similar syntax ?
```yaml
helmfile:
- path: helmfile1.yaml      # apply helmfile1
- - path: helmfile2a.yaml  # then helmfile2a & helmfile2b in parallel 
  - path: helmfile2b.yaml
```

**Comment by FACOLOMBANI** [comment]
Merge as @jouve proposed worked. I will have to customise the command to add missing configs, values and specify environment as expected. Thanks.

**Comment by yxxhero** [comment]
@jouve @FACOLOMBANI Thanks very much. I will close the issue. if has more questions. please ping me.
